Feat/automatic update installation

AdemolaAri opened the pull request:
## Summary
<!-- Describe your change -->
This is a feature request (#4322) to have goose auto-download and install updates when available. This change adds that functionality. 


### Type of Change
<!-- Select all that apply -->
- [x] Feature
- [ ] Bug fix
- [ ] Refactor / Code quality
- [ ] Performance improvement
- [ ] Documentation
- [ ] Tests
- [ ] Security fix
- [ ] Build / Release
- [ ] Other (specify below)

### Testing
<!-- How have this change been tested? Unit/integration tests? Manual testing? -->
Manual testing of the actual build. I added console.log statements and followed the follwing steps
1. Build the app. cd ui/desktop / npm run package / npm run make
2. Open the executable file and saw updates auto-downloaded to download folder
3. confirmed the right releases was auto-downloaded (tested on Mac only)

Note: because these builds are unsigned, i wasn't able to test the auto-install part. I believe that should work as well. 


### Related Issues
Relates to #ISSUE_ID  
Discussion: LINK (if any)

#4322



### Screenshots/Demos (for UX changes)
Before:  

<img width="2450" height="972" alt="image" src="https://github.com/user-attachments/assets/8310c84e-626a-4799-9350-e0a7dbb89a2d" />

<br/>

<br/>



After:   
<img width="2796" height="704" alt="image" src="https://github.com/user-attachments/assets/56c4e673-240a-416a-ba23-c6291f6839cd" />
|
|
<img width="2464" height="956" alt="image" src="https://github.com/user-attachments/assets/6a1a15fd-8a75-45b3-91da-f5a178df8397" />
|
|
<img width="2464" height="872" alt="image" src="https://github.com/user-attachments/assets/90da32de-1149-4be1-a433-ec09d3267778" />


<!-- For Recipe Cookbook Submissions ONLY: Include your email below to receive $10 OpenRouter credits once approved & merged -->
**Email**:

AdemolaAri commented on ui/desktop/src/utils/githubUpdater.ts:
```diff
@@ -106,7 +106,7 @@ export class GitHubUpdater {
       log.info(`GitHubUpdater: Available assets: ${release.assets.map((a) => a.name).join(', ')}`);
 
-      const asset = release.assets.find((a) => a.name === assetName);
+      const asset = release.assets.find((a) => a.name.toLowerCase() === assetName); // keeping comparisms to lower case becasue Goose vs goose
```
thought about either changing the hardcoded values in lines 94-103 or keeping this normalized comparison. Let me know if i should change the hardcoded values instead.

taniandjerry commented:
Thank you for your contribution! Let me add the tags, and tag teammates for review.

@DOsinga @alexhancock @michaelneale @jamadeo @zanesq @angelahning 

As we are in the weekend in AU and in the states, expect a review come Monday, ET! @AdemolaAri

AdemolaAri commented:
@taniandjerry , I'm sorry I've not been able to work on the other things I need for this - I do not have access to my computer until this Friday. It's fine if this doesn't count towards HacktoberFest. I plan to get the updates on this PR out sometime this weekend

taniandjerry commented:
> @taniandjerry , I'm sorry I've not been able to work on the other things I need for this - I do not have access to my computer until this Friday. It's fine if this doesn't count towards HacktoberFest. I plan to get the updates on this PR out sometime this weekend

That is amazing. Since your submission would likely pass beyond hacktoberfest, but you've been working on this feature with @zanesq that will make an impact on users for goose, I'll definitely be sure we are able to reward you in some way for still working on this. 🤗 ❤️ looking forward to getting this across the finish line!

zanesq commented:
@michaelneale made a [signed build](https://drive.google.com/file/d/1_uHjWf9-NwEJ6wNqXFayDXli7s7kf3_X/view?usp=sharing) to test from this branch but I get the following error message `Update Available but no download URL found for platform: darwin, arch: arm64` does that happen for you? 

<img width="650" height="172" alt="image" src="https://github.com/user-attachments/assets/4dc6cc56-8d63-4db9-86fc-6b0cd84a7857" />

AdemolaAri commented:
@zanesq , Now when you click the `Update Available` button, it navigates to the correct section of Setting page. 

Also, fixed `Update Available but no download URL found for platform: darwin, arch: arm64`, this is a Goose vs goose text issue. i had normalized the values during comparison but missed one .lowercase() addition. Added a unittest file to catch those types of issues going forward.

Please review and let me know :)

DOsinga commented on ui/desktop/src/utils/__tests__/githubUpdater.test.ts:
```diff
@@ -0,0 +1,341 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
```
delete this test - it doesn't test anything

AdemolaAri replied:
oh yeah, looking it at again it's all mocked stuff. I thought there was a part where it ran the full updater logic.

AdemolaAri commented:
> Thanks tested latest signed build from this branch and the download did work but the progress bar was jumpy until it finally downloaded (see below for recording)
> 
>  Kapture.2025-11-10.at.16.41.58.mp4 
> It didn't do the auto install but did open the downloads directory with the download.
> 
> Since its downloading now I think we can merge this after fixing the jumpy progress bar if you want to take a look at that then we can come back to the auto install in another PR later.

oh yeah, i have a plan to fix that if everything else checks out. I'll push a fix for that and we can plan the auto-install for another PR.

AdemolaAri commented:
@zanesq i'm back at this. sorry i was away for a few weeks.

Jumpy download bar should be fixed now. 

https://github.com/user-attachments/assets/008f0bdf-dd63-48b4-bdf5-0a41001a8d65

And I added the appropriate wording around auto-downloading but not auto-installing. I also added an 'open downloads and quit' action in the dialog to take users to the download folder and auto-quit the app removing one manual step at least.

https://github.com/user-attachments/assets/7f657666-2610-45b4-97fe-39e672a1df23

It does seem like auto-install works when the download is done thru the native electron path. The github fallback path seem to not be able to do auto-install for some reason. (I'm limited by not being able to do/test a signed build so just going by the code logic)

Pls test again and let me know how it goes.

zanesq reviewed (approved):
Verified LGTM thanks!
